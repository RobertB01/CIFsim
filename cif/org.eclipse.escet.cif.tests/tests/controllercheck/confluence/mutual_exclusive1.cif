//////////////////////////////////////////////////////////////////////////////
// Copyright (c) 2022 Contributors to the Eclipse Foundation
//
// See the NOTICE file(s) distributed with this work for additional
// information regarding copyright ownership.
//
// This program and the accompanying materials are made available
// under the terms of the MIT License which is available at
// https://opensource.org/licenses/MIT
//
// SPDX-License-Identifier: MIT
//////////////////////////////////////////////////////////////////////////////

automaton DifferentLocs:
  controllable a, b;
  uncontrollable u;

  location p:
    initial;
    edge a goto r;
    edge u goto q;
  location q:
    edge b goto r;
  location r:
    marked;
end
