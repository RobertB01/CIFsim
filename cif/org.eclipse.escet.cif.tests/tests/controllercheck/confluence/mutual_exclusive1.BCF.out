Loading CIF specification "controllercheck/confluence/mutual_exclusive1.cif"...
Preparing for BDD-based checks...

CIF variables and location pointers:
  Nr     Kind              Type  Name           Group  BDD vars  CIF values  BDD values  Values used
  -----  ----------------  ----  -------------  -----  --------  ----------  ----------  -----------
  1      location pointer  n/a   DifferentLocs  0      2 * 2     3 * 2       4 * 2       75%
  -----  ----------------  ----  -------------  -----  --------  ----------  ----------  -----------
  Total                                         1      4         6           8           75%

Skipping variable ordering: only one variable present.


Restricting system behavior using state/event exclusion plant invariants.

Preparing for MDD-based checks...
Analyzing automaton "DifferentLocs"...
    Initializing the automaton data for event "DifferentLocs.a"...
    Initializing the automaton data for event "DifferentLocs.b"...
    Processing edges from location "p" of automaton "DifferentLocs"...
    Processing edges from location "q" of automaton "DifferentLocs"...
    Processing edges from location "r" of automaton "DifferentLocs"...
    Updating global guards and updates for event "DifferentLocs.a"...
    Updating global guards and updates for event "DifferentLocs.b"...

Checking for bounded response...
Computing reachable states...
Reachable states: DifferentLocs.p [initial states predicate]
Forward reachability: iteration 1.
Reachable states: DifferentLocs.p -> not DifferentLocs.q [forward reach with edge: (event: DifferentLocs.a) (guard: DifferentLocs.p) (assignments: DifferentLocs := DifferentLocs.r)]
Reachable states: not DifferentLocs.q -> true [forward reach with edge: (event: DifferentLocs.u) (guard: DifferentLocs.p) (assignments: DifferentLocs := DifferentLocs.q)]
Forward reachability: iteration 2.
Reachable states: true [fixed point].

Computing bound for uncontrollable events...
Bounded response check round 1 (states before round: true).
Bounded response check round 2 (states before round: DifferentLocs.q).

Computing bound for controllable events...
Bounded response check round 1 (states before round: true).
Bounded response check round 2 (states before round: DifferentLocs.r).

Bounded response check completed.

Checking for finite response...

Iteration 1.

Checking for confluence...

Mutual exclusive event pairs:
    (DifferentLocs.a, DifferentLocs.b)

CONCLUSION:
    [OK] The specification has bounded response:
    
        - A sequence of at most 1 transition is possible for uncontrollable events.
        - A sequence of at most 1 transition is possible for controllable events.

    [OK] The specification has finite response.
    [OK] The specification has confluence.

The model with the check results has been written to "controllercheck/confluence/mutual_exclusive1.checked.cif".
